src/client/components/raw-data-modal/raw-data-modal.tsx

```diff
@@ -0,0 +1,312 @@
+      scrollTop: 0,
+      error: null
+    };
+    this.dimensionsWidths = [];
+    this.measuresWidths = [];
+  }
+
+  componentDidMount() {
+    this.mounted = true;
+    const { essence } = this.props;
+    this.fetchData(essence);
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  fetchData(essence: Essence): void {
+    const { dataSource } = essence;
+    const $main = $('main');
+    const query = $main.filter(essence.getEffectiveFilter(RawDataModal.id).toExpression()).limit(LIMIT);
+    this.setState({ loading: true });
+    dataSource.executor(query)
+      .then(
+        (dataset: Dataset) => {
+          if (!this.mounted) return;
+          this.setState({
+            dataset,
+            loading: false
+          });
+        },
+        (error: Error) => {
+          if (!this.mounted) return;
+          this.setState({
+            error,
+            loading: false
+          });
+        }
+      );
+  }
+
+  onScroll(e: UIEvent) {
+    const target = e.target as Element;
+    this.setState({
+      scrollLeft: target.scrollLeft,
+      scrollTop: target.scrollTop
+    });
+  }
+
+  getStringifiedFilters(): List<string> {
+    const { essence } = this.props;
+    const { dataSource } = essence;
+    return essence.getEffectiveFilter(RawDataModal.id).clauses.map((clause, i) => {
+      const dimension = dataSource.getDimensionByExpression(clause.expression);
+      if (!dimension) return null;
+      return formatLabel({ dimension, clause, essence, verbose: true });
+    }).toList();
+  }
+
+  renderFilters(): List<JSX.Element> {
+    const stringifiedFilters = this.getStringifiedFilters();
+    var filters = stringifiedFilters.map((filter: string, i: number) => {
+      return <li className="filter" key={i}>{filter}</li>;
+    });
+    filters = filters.concat(<li className="filter" key="filter">limit: {LIMIT}</li>);
+    return filters.toList();
+  }
+
+  renderHeader(): JSX.Element[] {
+    const { essence } = this.props;
+    const { dataSource } = essence;
+    const dimensions = dataSource.dimensions;
+    const measures = dataSource.measures;
+    const timeAttribute: string = dataSource.timeAttribute.name;
+    const timeColStyle = { width: TIME_COL_WIDTH };
+    var cols = [ <div className={thClassName} style={timeColStyle} key="time">
+      <span className={thText}>
+        {makeTitle(timeAttribute)}
+      </span>
+    </div> ];
+    this.dimensionsWidths = [TIME_COL_WIDTH];
+    dimensions.map((dimension, i) => {
+      if (dimension.name === timeAttribute) return;
+      const width = getColumnWidth(dimension.kind);
+      const style = { width };
+      const key = dimension.title;
+      this.dimensionsWidths = this.dimensionsWidths.concat(width);
+      cols = cols.concat(<div className={thClassName} style={style} key={i}>
+        <div className={thText}>
+          {key}
+        </div>
+      </div>);
+    });
+
+    this.measuresWidths = [];
+    measures.map((measure, i) => {
```

Comment:
this is fundamentally wrong. This dialog is doing a raw (select) query. measures and dimensions do not exist here. only attributes (columns). For dimensions we logically could add the derived dimensions to the query (only once Druid goes to 0.9.1 though) but measures only exist when there is a split. The string "measure" should not exist in this file.